src/sentry/api/endpoints/project_artifact_bundle_files.py

```diff
@@ -0,0 +1,111 @@
+import base64
+from typing import List, Tuple
+
+from django.utils.functional import cached_property
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint, ProjectReleasePermission
+from sentry.api.paginator import ChainPaginator
+from sentry.api.serializers import serialize
+from sentry.constants import MAX_ARTIFACT_BUNDLE_FILES_OFFSET
+from sentry.models import ArtifactBundle, ArtifactBundleArchive, ProjectArtifactBundle
+from sentry.ratelimits.config import SENTRY_RATELIMITER_GROUP_DEFAULTS, RateLimitConfig
+
+
+class ArtifactBundleSource:
+    def __init__(self, files: dict):
+        self._files = files
+
+    @cached_property
+    def sorted_and_filtered_files(self) -> List[Tuple[str, dict]]:
+        files = [(file_path, info) for file_path, info in self._files.items()]
+        files.sort(key=lambda item: item[0])
+
+        return files
+
+    def __len__(self):
+        return len(self.sorted_and_filtered_files)
+
+    def __getitem__(self, range):
+        return self.sorted_and_filtered_files[range]
+
+
+@region_silo_endpoint
+class ProjectArtifactBundleFilesEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectReleasePermission,)
+    rate_limits = RateLimitConfig(
+        group="CLI", limit_overrides={"GET": SENTRY_RATELIMITER_GROUP_DEFAULTS["default"]}
+    )
+
+    def get(self, request: Request, project, bundle_id) -> Response:
+        """
+        List a Project Artifact Bundle's Files
```

Comment:
```suggestion
        List files for a given project artifact bundle
```

Maybe more readable this way?